libraries/AP_CANManager/AP_CANSensor.cpp

```diff
@@ -22,21 +22,53 @@
 #include <AP_Scheduler/AP_Scheduler.h>
 #include "AP_CANSensor.h"
 
+#ifdef HAL_BUILD_AP_PERIPH
+#include <../Tools/AP_Periph/AP_Periph.h>
+#endif
+
 extern const AP_HAL::HAL& hal;
 
+#if HAL_ENABLE_LIBUAVCAN_DRIVERS
 #define debug_can(level_debug, fmt, args...) do { AP::can().log_text(level_debug, _driver_name, fmt, ##args); } while (0)
+#else
+#define debug_can(level_debug, fmt, args...)
+#endif
 
 CANSensor::CANSensor(const char *driver_name, AP_CANManager::Driver_Type dtype, uint16_t stack_size) :
     _driver_name(driver_name),
     _stack_size(stack_size)
 {
+#if HAL_ENABLE_LIBUAVCAN_DRIVERS
     if (!AP::can().register_driver(dtype, this)) {
         debug_can(AP_CANManager::LOG_ERROR, "Failed to register CANSensor %s", driver_name);
     } else {
         debug_can(AP_CANManager::LOG_INFO, "%s: constructed", driver_name);
     }
+#elif defined(HAL_BUILD_AP_PERIPH)
+    register_driver_ap_periph(dtype);
+#endif
 }
 
+#ifdef HAL_BUILD_AP_PERIPH
+void CANSensor::register_driver_ap_periph(const AP_CANManager::Driver_Type dtype)
+{
+    for (uint8_t i = 0; i < HAL_NUM_CAN_IFACES; i++) {
+        if (AP::periph().can_protocol(i) != dtype) {
```

Comment:
this was supposed to be removed?

Reply:
Sid and I had a rebase/force mixup. I overwrite his fix, my bad. All fixed now.